### ISSUE TITLE
Unit convertor breaks framework when no units are available.

### ISSUE THREAD

**Issue by ahmedfiaz** [NOAA-GFDL/MDTF-diagnostics#154]
I'm currently testing v3.0 on CMIP6 models. Several model outputs do not have units for variables such as `lat_bnds` and `lon_bnds`. This breaks the framework because this assertion statement here fails.
https://github.com/NOAA-GFDL/MDTF-diagnostics/blob/70622b2fb9395de3a8ac3bc92c55c955e3d489b2/src/units.py#L126
which gives the error:
```
WARNING: Execution of POD Wheeler_Kiladis couldn't be completed:
***** Caught exception #1:
Traceback (most recent call last):
  File "/home/fiaz/MDTF_v3/src/preprocessor.py", line 578, in process
    ds = f.process(var, ds)
  File "/home/fiaz/MDTF_v3/src/preprocessor.py", line 291, in process
    ds = units.convert_dataarray(ds, c.bounds, dest_c.units)
  File "/home/fiaz/MDTF_v3/src/units.py", line 126, in convert_dataarray
    assert src_unit
AssertionError
```
Commenting out the assertion statement allows the framework to run fine. 

Should we make the framework more forgiving with missing units or include a warning in the docs?

**Comment by wrongkindofdoctor** [comment]
@ahmedfiaz @jkrasting @tsjackson-noaa I don't know about permitting missing units, since we want to enforce CF-compliant data. I do think we should note in the documentation that files with non-CF-compliant metadata may cause errors, and that users may need to modify their input data files accordingly.

**Comment by ahmedfiaz** [comment]
@tsjackson-noaa The CMIP6 file that I used for testing the Wheeler Kiladis POD here:
https://www.dropbox.com/sh/pfaezwq2npboizv/AAAnId8GA_QBBS2ia9BsqHWwa?dl=0

**Comment by tsjackson-noaa** [comment]
Closing this issue as resolved by PR #161.
